src/librustc/metadata/cstore.rs

```diff
@@ -59,11 +60,17 @@ pub struct crate_metadata {
     pub name: String,
     pub local_path: RefCell<SmallVector<ast_map::PathElem>>,
     pub data: MetadataBlob,
-    pub cnum_map: cnum_map,
+    pub cnum_map: RefCell<cnum_map>,
     pub cnum: ast::CrateNum,
     pub codemap_import_info: RefCell<Vec<ImportedFileMap>>,
     pub span: codemap::Span,
-    pub staged_api: bool
+    pub staged_api: bool,
+
+    /// Flag if this crate is required by an rlib version of this crate, or in
+    /// other words whether it was explicitly linked to. An example of a crate
+    /// where this is false is when an allocator crate is injected into the
+    /// dependency list, and therefore isn't actually needed to link an rlib.
+    pub required_by_rlib: Cell<bool>,
```

Comment:
I find the name of this variable and the comments hard to understand, even as somebody who knows what this current patch does. I think this flag just means 'not an allocator crate', but I'm not sure.